addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -257,6 +250,30 @@ export class SelfOrder extends Reactive {
         }
     }
 
+    updateCategoryList() {
+        let now = luxon.DateTime.now();
+        now = now.hour + now.minute / 60;
+        const availableCategories = this.pos_category
+            .sort((a, b) => a.sequence - b.sequence)
+            .filter((c) => this.productsGroupedByCategory[c.id])
+            .sort((a, b) => categorySorter(a, b, this.config.iface_start_categ_id));
+        this.categoryList = new Set(availableCategories);
+
+        this.availableCategoryListIds = availableCategories
+            .filter((c) => {
+                if (c.hour_after > now || c.hour_until < now) {
+                    return false;
+                }
+                return true;
```

Comment:
```suggestion
                return now > c.hour_after && now < c.hour_until
```

Or `return !(c.hour_after > now || c.hour_until < now)`

Reply:
Yep that's better